packages/astro/src/assets/utils/transformToPath.ts

```diff
@@ -5,7 +5,29 @@ import { shorthash } from '../../runtime/server/shorthash.js';
 import type { ImageTransform } from '../types.js';
 import { isESMImportedImage } from './imageKind.js';
 
-export function propsToFilename(filePath: string, transform: ImageTransform, hash: string) {
+/**
+ * Converts a file path and transformation properties of the transformation image service, into a formatted filename.
+ *
+ * The formatted filename follows this structure:
+ *
+ * `<prefixDirname>/<baseFilename>_<hash><outputExtension>`
+ *
+ * - `prefixDirname`: If the image is an ESM imported image, this is the directory name of the original file path; otherwise, it will be an empty string.
+ * - `baseFilename`: The base name of the file or a hashed short name if the file is a `data:` URI.
+ * - `hash`: A unique hash string generated to distinguish the transformed file.
+ * - `outputExtension`: The desired output file extension derived from the `transform.format` or the original file extension.
+ *
+ * ## Example
```

Comment:
Is there a reason to use a Markdown title here instead of the JSDoc `@example` tag?
```suggestion
 * @example
```

Reply:
Different rendering. The `## Example` is markdown, so it gets rendered as a `h2`, but `@example` renders differently, all the after it is rendered in a code block, which isn't what we want here:

![Screenshot 2025-03-10 at 09 36 09](https://github.com/user-attachments/assets/cbe566a3-0957-4c46-8b8d-e70d556507a0)